lib/engine/game/g_1849/step/buy_e_token.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1849
+      module Step
+        class BuyEToken < Engine::Step::Base
+          def actions(entity)
+            actions = super.dup
+            actions += %w[choose pass] if can_buy_e_token?(entity)
+
+            actions.uniq
+          end
+
+          def description
+            'Buy E-Token'
+          end
+
+          def pass_description
+            'Pass'
+          end
+
+          def choices
+            choices = []
+            choices << ["Buy E-Token for #{@game.format_currency(e_token_cost)}"] if can_buy_e_token?(current_entity)
+            choices
+          end
+
+          def choice_name
+            'E-Token allows the purchase of E-Trains'
+          end
+
+          def log_skip(entity)
+            super if @game.electric_dreams? && @game.phase.status.include?('gray_uses_black')
```

Comment:
Use some other check that is connected to electric dreams, not another status that just happens to line up.